### PR TITLE
Improve handling of symlinks to szcc

### PR DESCRIPTION
**Pull request by Dead2**
Cmake is now making clang/clang++ symlinks in the build dir, but installs them to the doc dir.
That way the user can prepend either of those to PATH env to have a buildsystem auto-detect them as clang.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -85,6 +87,8 @@ install(
 
 install(
   FILES LICENSE
+  ${CMAKE_CURRENT_BINARY_DIR}/clang
+  ${CMAKE_CURRENT_BINARY_DIR}/clang++
```
Comment (magras) [inline CMakeLists.txt]: Where this symlinks will point after installation?

Comment (Dead2) [reply CMakeLists.txt]: Good question. The intention is to have them link to the main bin/szcc file, but I am uncertain about how cmake handles the links.

Comment (magras) [reply CMakeLists.txt]: I've checked and looks like cmake installs a copy of the symlink target. At least that's what I've got for `szcc++`:
```
# ls -l /usr/local/bin
total 24
-rwxr-xr-x 1 root root 10174 May 15 23:21 szcc
-rwxr-xr-x 1 root root 10174 May 15 23:21 szcc++
```

Comment (Dead2) [reply CMakeLists.txt]: It actually didn't copy the files, but it made a hardlink, so not at all what was expected. Fixed in new commit.

**Review by magras — APPROVED** [review]
I'm sorry, I missed this PR somehow.
Looks good to me.

**Pull request merged**